### PR TITLE
Deckbuilding fixes and improvements

### PR DESCRIPTION
**Pull request by StephenGibbsDev**
-added deck selector
-added functionality to add to selected deck
-added unit test for decklist

### CODE REVIEW

File: src/MTGCore.ClientApp/ClientApp/src/App.vue [hunk 1]
```diff
@@ -29,14 +29,24 @@
                   </div>
                   <small id="emailHelp" class="form-text text-muted">Lorem Ipsum Dolar set ammet</small>
                 </div>
-                <ResultsTable v-bind:post="post" />
+                <ResultsTable v-on:addCardToDeck="addToDeck" v-bind:post="post" />
```
Comment (StephenGibbsDev) [inline src/MTGCore.ClientApp/ClientApp/src/App.vue]: this listens for a "addCardToDeck" event which is emmited in a child component. 

best practice for vuejs is passing data down to child via props and from child to parent using events.

Comment (ConnorDownes) [reply src/MTGCore.ClientApp/ClientApp/src/App.vue]: Is the addCardToDeck just a custom event name? I understand addToDeck is the method below.

File: src/MTGCore.ClientApp/ClientApp/src/components/ResultsTable.vue [hunk 2]
```diff
@@ -40,20 +35,7 @@ export default {
...
-        .then(() => {
-          this.someMethod();
-        })
-        .catch(err => {
-          alert(`There was an error submitting your form. See details: ${err}`);
-        });
-    },
-    someMethod(){
-        this.$parent.updateDeckList();
+      this.$emit("addCardToDeck", id);
```
Comment (ConnorDownes) [inline src/MTGCore.ClientApp/ClientApp/src/components/ResultsTable.vue]: So is this calling the function from the parent App.vue?

File: src/MTGCore/API/DeckController.cs [hunk 3]
```diff
@@ -29,27 +29,24 @@ public DeckController(IRepoContext context, MTGService mtgservice, IMapper mappe
         }
 
         //api/deck/
-        [Route("Add/{deckID}/{multiverseID}")]
+        [Route("Add/{deckID}/{cardID}")]
         [HttpPost]
-        public async Task Post(int deckID, int multiverseID)
+        public async Task Post(int deckID, string cardID)
         {
```
Comment (ConnorDownes) [inline src/MTGCore/API/DeckController.cs]: Is there any reason why the cardID is a string? Does the API return it as a string?

File: src/MTGCore/Services/MTGService.cs [hunk 4]
```diff
@@ -35,6 +35,21 @@ public async Task<Card> GetCardByID(int multiverseID)
...
+        {
+            var result = await _client.GetAsync($"cards/{Id}");
+
+            if (!result.IsSuccessStatusCode)
+                return default;
+
+            var stream = await result.Content.ReadAsStringAsync();
+
+            var singleCard = JsonConvert.DeserializeObject<RootObject>(stream).card;
+
```
Comment (ConnorDownes) [inline src/MTGCore/Services/MTGService.cs]: Would it be worth creating a deserialiser class and injecting it? So if a different API were used in the future which returned XML, a new implementation of the deserialiser could be created and pass in that instead?

Comment (ConnorDownes) [reply src/MTGCore/Services/MTGService.cs]: Also naming convention for 'RootObject' could probably be better but it's a minor thing (I know that was probably something from Json2C#)

File: src/MTGCore.ClientApp/ClientApp/src/components/ResultsTable.vue (hunk 2 above)
Comment (StephenGibbsDev) [reply src/MTGCore.ClientApp/ClientApp/src/components/ResultsTable.vue]: not really calling. Its creating an event with the name addCardToDeck with an ID value. Then the v:on command is used basicalled to say "whenever you see the addCardToDeck event run that named function"

File: src/MTGCore.ClientApp/ClientApp/src/App.vue (hunk 1 above)
Comment (StephenGibbsDev) [reply src/MTGCore.ClientApp/ClientApp/src/App.vue]: i think so...
https://vuejs.org/v2/guide/components-custom-events.html#Event-Names
it says to use kebab case as well so i wiill change

File: src/MTGCore/API/DeckController.cs (hunk 3 above)
Comment (StephenGibbsDev) [reply src/MTGCore/API/DeckController.cs]: not sure, will try with int and see if it works

File: src/MTGCore/Services/MTGService.cs (hunk 4 above)
Comment (StephenGibbsDev) [reply src/MTGCore/Services/MTGService.cs]: > Would it be worth creating a deserialiser class and injecting it? So if a different API were used in the future which returned XML, a new implementation of the deserialiser could be created and pass in that instead?

Excellent idea. Will create as issue and add to the TODO list.

**Pull request merged**